Reemit.Decompiler.Clr/Methods/FatMethodHeader.cs

```diff
@@ -0,0 +1,24 @@
+namespace Reemit.Decompiler.Clr.Methods;
+
+public class FatMethodHeader(CorILMethodFlags flags, byte size, ushort maxStack, uint codeSize, uint localVarSigTok)
+    : IMethodHeader
+{
+    public CorILMethodFlags Flags { get; } = flags;
+    public byte Size { get; } = size;
+    public ushort MaxStack { get; } = maxStack;
+    public uint CodeSize { get; } = codeSize;
+    public uint LocalVarSigTok { get; } = localVarSigTok;
+
+    public static FatMethodHeader Read(BinaryReader reader)
+    {
+        var flagsAndSize = reader.ReadUInt16();
+        const int flagsMask = (1 << 12) - 1;
+        var flags = (CorILMethodFlags)(flagsAndSize & flagsMask);
+        var size = (byte)(flagsAndSize >> 12);
```

Comment:
Maybe my reading of the spec is wrong, but are you sure this is correct? From the spec:

![image](https://github.com/v0idzz/Reemit/assets/7999934/f17ce2db-1032-4a3d-b550-f10ab402cea5)

This is my first time reading this, but it seems to me the table is saying the entirety of the first byte goes to flags, as well as the 4 most significant bits of the second byte. Then, the 4 least significant bits of the second byte are size. However, `BinaryReader` is little-endian, so `ReadUInt16` reverses them.

To test this, I wrote this small piece of code: 

![image](https://github.com/v0idzz/Reemit/assets/7999934/7f919ef4-b4c7-4bc2-ba46-d3ca876c90e7)

Maybe I'm getting confused here, but given the following bytes:

```cs
            [
                0b11111111,
                0b11110000,
            ];
```

I would expect `flags` to be `0b1111_11111111` (note the little-endianness still applies) and `size` `0b0000`, but the result is otherwise. Thoughts? I'm going purely off the words/tables of the spec, which definitely leaves something to be desired.

Reply:
Resolving because it sounds like the current implementation is correct.